### PR TITLE
Issue/python lower bound

### PR DESCRIPTION
**Pull request by wouterdb**
# Description

1. Move python to 3.12
2. Update to be in-line with ISO: I backported a LOT, I hope it doesn't catch fire 

closes https://github.com/inmanta/inmanta/issues/137

# Self Check:

- [x] Attached issue to pull request
- [x] Changelog entry

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -19,17 +19,8 @@ public = true
...
 
-[tool.irt.publish.docker_image.dev]
-image_registry = "code.inmanta.com:4567"
-image_repository = "solutions/containers"
-image_name = "orchestrator"
-image_tags = [
-    "dev",
-]
-registry_username_env = "GITLAB_REGISTRY_USERNAME"
-registry_password_env = "GITLAB_REGISTRY_PASSWORD"
```
Comment (arnaudsjs) [inline pyproject.toml:30]: If we move this to the native image, we need to update the documentation as well.

File: changelogs/unreleased/Update_python_version_for_iso8 copy.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+description: Update python lower bound to 3.12 for ISO8
+change-type: major
+destination-branches: [master]  
+sections:
+  feature: "{{description}}"
```
Comment (arnaudsjs) [inline changelogs/unreleased/Update_python_version_for_iso8 copy.yml:5]: We must also mention that we use the native docker containers.

File: inmanta.spec [hunk 3]
```diff
@@ -265,24 +241,6 @@ if [ -e "/etc/inmanta/server.cfg" ]; then
...
-done
-# Rename all (autostarted) agent venvs for which the python version is
-# out-of-date with the python version of the server venv.
-for server_env in /var/lib/inmanta/*; do
-  agent_venv_dir="${server_env}/agent/env"
-  if [ -e "${agent_venv_dir}/bin" ] && [ ! -e "${agent_venv_dir}/bin/${python_version}" ]; then
-    mv "${agent_venv_dir}" "${agent_venv_dir}.rpmsave_$(date +'%%Y%%m%%d_%%H%%M%%S')"
-  fi
-done
-
```
Comment (arnaudsjs) [inline inmanta.spec:285]: We also have to remove the script that migrates the venv of the manually started agents.

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: setup.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 
 setup(
     version="2024.4.1",
-    python_requires=">=3.11",  # also update classifiers
+    python_requires=">=3.12",  # also update classifiers
```
[the remark this answers is not in the document]
Comment (wouterdb) [reply setup.py:6]: Where? I couldn't find them?

File: changelogs/unreleased/Update_python_version_for_iso8 copy.yml
[the remark this answers is not in the document]
Comment (wouterdb) [reply changelogs/unreleased/Update_python_version_for_iso8 copy.yml:1]: right, copy-paste mistake, thx

File: setup.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (arnaudsjs) [reply setup.py:6]: This `setup()` method has a keyword argument `classifiers`

File: changelogs/unreleased/Update_python_version_for_iso8 copy.yml (hunk 2 above)
Comment (wouterdb) [reply changelogs/unreleased/Update_python_version_for_iso8 copy.yml:5]: Then I will need two changelog files, I presume?

File: inmanta.spec (hunk 3 above)
Comment (wouterdb) [reply inmanta.spec:285]: thx

File: changelogs/unreleased/Update_python_version_for_iso8 copy.yml (hunk 2 above)
Comment (wouterdb) [reply changelogs/unreleased/Update_python_version_for_iso8 copy.yml:5]: Can I defer this to https://github.com/inmanta/inmanta-core/issues/8304

File: pyproject.toml (hunk 1 above)
Comment (wouterdb) [reply pyproject.toml:30]: Can I defer this to https://github.com/inmanta/inmanta-core/issues/8304 as well?

File: docker/native_image/Dockerfile [hunk 5]
```diff
@@ -91,5 +91,5 @@ WORKDIR /var/lib/inmanta
 HEALTHCHECK --interval=10s --timeout=3s --start-period=5m --start-interval=5s --retries=3 \
     CMD /opt/inmanta/bin/python -c "from inmanta import config as c, protocol as p; c.Config.load_config(config_dir='/etc/inmanta/inmanta.d'); r = p.SyncClient('client').get_server_status(); assert r.code == 200 and {s['name']: s for s in r.result['data']['slices']}['core.database']['status']['connected'], str(r.result)"
 
-ENTRYPOINT ["/usr/bin/inmanta"]
+ENTRYPOINT ["/usr/bin/tini", "--", "/usr/bin/inmanta"]
```
Comment (edvgui) [inline docker/native_image/Dockerfile:94]: This resolves this: https://github.com/inmanta/inmanta/issues/137

**Review by edvgui — APPROVED** [review]
Lgtm, I focused on the container side

File: changelogs/unreleased/Update_python_version_for_iso8 copy.yml (hunk 2 above)
Comment (arnaudsjs) [reply changelogs/unreleased/Update_python_version_for_iso8 copy.yml]: You can have a `feature` and an `upgrade-note` section in one changelog file. But I am fine with deferring this to the referenced ticket.

File: setup.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (arnaudsjs) [reply setup.py:6]: You still need to handle this comment.

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

**Review by arnaudsjs — APPROVED** [review]

File: changelogs/unreleased/update_python_version_for_iso8.yml
Comment (Hugo-Inmanta) [inline changelogs/unreleased/update_python_version_for_iso8.yml:1]: Why 2 changelog files instead of 2 sections in the same file ?

**Review by Hugo-Inmanta — APPROVED** [review]

**Pull request closed without merging**